### PR TITLE
Log incoming

### PR DESCRIPTION
**Pull request by jofomah**
note there are some todos which will be dependent on the IDB etc. but this is combination of my branch and the master.

### CODE REVIEW

File: app/Ebro/js/lib/TouchSwipe/demos/Page_zoom.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <html>
     <head>  
     	<!-- 
-    		NOTE viewport is set so user can scale 
+    		NOTE viewport is set so user can scale
```
Comment (tlvince) [inline app/Ebro/js/lib/TouchSwipe/demos/Page_zoom.html:5]: Looks like your editor/IDE has removed trailing whitespace? I think we should treat Ebro as read-only vendor code.

**Comment by tlvince** [comment]
@jofomah, for future reference, it's considered good form to use a <= 50 character commit subject and use the body for expansion. See [A Note About Git Commit Messages](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).

**Comment by jofomah** [comment]
ok, cool

File: app/index.html [hunk 2]
```diff
@@ -9,9 +9,9 @@
   <!-- Place favicon.ico and apple-touch-icon.png in the root directory -->
   <!-- build:css styles/vendor.css -->
   <!-- bower:css -->
+  <link rel="stylesheet" href="bower_components/bootstrap/dist/css/bootstrap.css" />
```
Comment (tlvince) [inline app/index.html]: Looks like your `bower.json` is out of date, as in #52, we switched to a CSS-only version of bootstrap.

File: app/scripts/controllers/batches.js [hunk 3]
```diff
@@ -2,6 +2,17 @@
 
 angular.module('lmisChromeApp')
 /**
+ * configure routes for batch module
+ */
+    .config(function ($stateProvider) {
+      $stateProvider
+          .state('batchListView', {
+            url: '/batch-list-view',
```
Comment (tlvince) [inline app/scripts/controllers/batches.js:10]: We don't necessarily need to define a state URL as we're targeting Chrome apps which hide the address bar. They're still needed when you want to use `$stateParams` though.

Comment (jofomah) [inline app/scripts/controllers/batches.js:10]: well, i guess you dont that know most of this scripts will be reused for
the web based app, so then we will need it

File: app/scripts/controllers/bundle.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+var chromeApp = angular.module('lmisChromeApp');
```
Comment (tlvince) [inline app/scripts/controllers/bundle.js]: This module definition style pollutes the global namespace. It's better to use the "fluent" style, see [Modularizing AngularJS Applications](http://henriquat.re/modularizing-angularjs/modularizing-angular-applications/modularizing-angular-applications.html) (Fluent Definition of Modules)

File: app/scripts/controllers/batches.js (hunk 3 above)
Comment (tlvince) [inline app/scripts/controllers/batches.js:10]: OK, in that case, add them in.

File: app/scripts/controllers/bundle.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  $scope.logIncomingForm = {};
+  $scope.logIncomingForm.verify = [];
+  $scope.logIncomingForm.storage_units = [];
+
+  $scope.getCurrentDate = function () {
+    var today = new Date();
+    return $filter('date')(today, "yyyy-MM-dd");
+  }
+
+  userFactory.getLoggedInUser().then(function (data) {
```
Comment (tlvince) [inline app/scripts/controllers/bundle.js]: Resolved data is better placed in the state's [resolve map](https://github.com/angular-ui/ui-router/wiki#wiki-resolve) and injected into the controller. That method allows us to test the controller easier (as the injections can be mocked).

File: app/views/batches/index.html [hunk 6]
```diff
@@ -1,8 +1,7 @@
 <div class="row">
   <ul class="breadcrumb">
-    <li><a href="#/main/"><i class="fa fa-home"> Home</i></a></li>
-    <li><a href="#/products/products">Product</a></li>
-    <li class="active">Product Item</li>
+    <li><a ui-sref="home.index.controlPanel"><i class="fa fa-home"> Home</i></a></li>
```
Comment (tlvince) [inline app/views/batches/index.html:3]: Don't worry about updating the breadcrumbs; they'll be generated based on the current state (and its parents) using [tv-breadcrumbs](http://github.com/tlvince/tv-breadcrumbs).

File: app/views/bundles/incoming-log.html [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+      <li> Bundle</li>
+      <li class="active">Log Incoming</li>
+    </ol>
+  </div>
+
+  <div class="row" ng-show="!show">
+    <div class="row alert-danger" ng-if="clicked && !found">
+      <div class="col-sm-18">
+        <div class="col-sm-12">
+          <h5>Bundle with Bundle No: {{ showBundleNo }} not found.</h5>
```
Comment (tlvince) [inline app/views/bundles/incoming-log.html:14]: To support localisation, I've been adding all new strings to [messages.json](https://github.com/eHealthAfrica/LMIS-Chrome/blob/c66e25c69b83879f263730fb932c4b52d7398c9f/app/_locales/en/messages.json), which can be referenced using [i18n('key')](https://github.com/eHealthAfrica/LMIS-Chrome/blob/c66e25c69b83879f263730fb932c4b52d7398c9f/app/scripts/app.js#L39-L40). What do you think?

**Comment by jofomah** [comment]
i have not implemented tv-breadcrumb on all my views (thats why i said there are a lot of pending codes but i want our branches to stay up to date) cause my branch is already way ahead of the master

File: app/views/bundles/incoming-log.html (hunk 7 above)
Comment (jofomah) [inline app/views/bundles/incoming-log.html:14]: yeah i noticed and even added mine, to message.json, still replacing them through the views

**Comment by tlvince** [comment]
You can avoid superfluous merge commits (e.g. `Merge remote-tracking branch 'origin/master' into log-incoming`) by using rebase `git pull --rebase origin master` (or `git checkout origin master; git pull origin master; git checkout [branch]; git rebase master`).

File: app/views/bundles/incoming-log.html (hunk 7 above)
Comment (tlvince) [inline app/views/bundles/incoming-log.html:14]: Cool, although, we may need to rethink this if/when the app is ported for use outside of Chrome apps.

File: app/scripts/controllers/bundle.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+angular.module('lmisChromeApp').chromeApp.config(function ($stateProvider) {
```
Comment (tlvince) [inline app/scripts/controllers/bundle.js]:
``` diff
-angular.module('lmisChromeApp').chromeApp.config(function ($stateProvider) {
+angular.module('lmisChromeApp').config(function ($stateProvider) {
```

**Comment by tlvince** [comment]
Would you mind squashing the last three commits?

**Comment by jofomah** [comment]
ce4a668  where i made the correction.

**Pull request merged**